src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx

```diff
@@ -0,0 +1,204 @@
+  const [showError, setShowError] = useState(false)
+
+  const handleOrgNameChange = useCallback(
+    e => {
+      changeOrgName(e.target.value)
+      setShowError(false)
+      setInputError(DEFAULT_ERROR_MESSAGE)
+    },
+    [changeOrgName]
+  )
+
+  useEffect(() => {
+    if (createOrgLoadingStatus === RemoteDataState.Error) {
+      if (error === 'NAME_CONFLICT') {
+        setShowError(true)
+        setInputError(NAME_CONFLICT_MESSAGE)
+      } else {
+        setErrorMessage(errorMap?.[error])
+      }
+    } else {
+      setShowError(false)
+    }
+  }, [error, createOrgLoadingStatus])
+
+  const handleCreateOrg = useCallback(() => {
+    if (orgName.trim() === '') {
+      setShowError(true)
+      setInputError(DEFAULT_ERROR_MESSAGE)
+      return
+    }
+
+    createOrg()
+      .then(() => {
+        onClose()
+        dispatch(notify(orgCreateSuccess()))
+      })
+      .catch(e => {
+        if (e instanceof UnauthorizedError) {
+          setError('UNAUTHORIZED')
+        } else if (e instanceof ServerError) {
+          setError('SERVER_ERROR')
+        } else if (e instanceof UnprocessableEntityError) {
+          // Even though there can be different meanings/reasons for this issue.
+          //  We are deliberately choosing to display this error(Status Code: 422)
+          //  as Duplicate Org Name issue.
+          setError('NAME_CONFLICT')
+        } else {
+          setError('GENERIC_ERROR')
+        }
+        changeCreateOrgLoadingStatus(RemoteDataState.Error)
+      })
+  }, [
+    orgName,
+    createOrg,
+    onClose,
+    dispatch,
+    changeCreateOrgLoadingStatus,
+    setError,
+  ])
+
+  const hasClusters = !!Object.keys(clusters).length
+  const cancelButtonStatusMap = {
+    [RemoteDataState.Loading]: ComponentStatus.Disabled,
+    [RemoteDataState.Error]: ComponentStatus.Default,
+    [RemoteDataState.Done]: ComponentStatus.Default,
+    [RemoteDataState.NotStarted]: ComponentStatus.Default,
+  }
+  const createButtonStatusMap = {
+    [RemoteDataState.Loading]: ComponentStatus.Loading,
+    [RemoteDataState.Error]: ComponentStatus.Disabled,
+    [RemoteDataState.Done]: ComponentStatus.Default,
+    [RemoteDataState.NotStarted]: ComponentStatus.Default,
+  }
+  return (
+    <Overlay.Container
+      maxWidth={750}
+      className="create-org-overlay-container"
+      testID="create-org-overlay--container"
+    >
+      <Overlay.Header
+        testID="create-org-overlay-header"
+        title="Create an Organization"
+        onDismiss={onClose}
+      />
+      <Overlay.Body>
+        {errorMessage && (
+          <Alert
+            style={errorStyle}
+            color={ComponentColor.Danger}
+            icon={IconFont.AlertTriangle}
+          >
+            {errorMessage}
+          </Alert>
+        )}
+        <FlexBox
+          direction={FlexDirection.Column}
+          className="create-org-overlay-body"
+        >
+          <InputWithLabel
```

Comment:
What does InputWithLabel do? 

We have `Form` and it's `Form.ValidationElement` that handles the displaying of errors, perhaps that will do here?

Reply:
I started off with the ValidationElement but it requires a `validationFunction` and does not provide the flexibility to display an error based on your requirement/need. If you know that component better, then I'd be happy to change it back.